analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Branching.LearnConstraint.cs

```diff
@@ -632,6 +632,38 @@ public void Branching_LearnsObjectConstraint_IsNull()
                 .And.ContainSingle(x => x.HasConstraint(ObjectConstraint.NotNull));
         }
 
+        [DataTestMethod]
+        [DataRow("arg.HasValue")]
+        [DataRow("arg.HasValue is true")]
+        [DataRow("arg.HasValue == true")]
+        [DataRow("arg.HasValue != false")]
+        [DataRow("!!arg.HasValue")]
+        public void Branching_LearnsObjectConstraint_Nullable_HasValue_True(string expression)
+        {
+            var validator = CreateIfElseEndValidatorCS(expression, OperationKind.DeclarationPattern, "int?");
+            validator.ValidateTag("If", x => x.Should().HaveOnlyConstraint(ObjectConstraint.NotNull));
+            validator.ValidateTag("Else", x => x.Should().HaveOnlyConstraint(ObjectConstraint.Null));
+            validator.TagValues("End").Should().HaveCount(2)
+                .And.ContainSingle(x => x.HasConstraint(ObjectConstraint.Null))
+                .And.ContainSingle(x => x.HasConstraint(ObjectConstraint.NotNull));
```

Comment:
NIT: We also used `SaftifyRespectivly` in such cases, which seems more appropriate.